### PR TITLE
feat: add FluxRecord.row with response data stored in Array

### PR DESCRIPTION
**Pull request by michaelahojna**
## Proposed Changes

Adding possibility of accessing response data in Array FluxRecord.row.

In case of using pivot on data, where field contains labels that occur by default in the annotated CSV (f.e. "result" or "table"), could be duplicated column names in response. In that case FluxRecord.values, which hold only unique keys, doesn't show complete data. This edge case is solved by using FluxRecord.row.

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] `swift test` completes successfully
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

File: Sources/InfluxDBSwift/FluxCSVParser.swift [hunk 1]
```diff
@@ -123,41 +127,58 @@ internal class FluxCSVParser {
...
+                  """)
+        }
     }
 
     private func parseRow(row: [String]) throws -> QueryAPI.FluxRecord {
         guard let table = table else {
             throw InfluxDBClient.InfluxDBError.generic(Self.errorMessage)
         }
-
+        var recordRow: [Any] = Array()
```
Comment (bednar) [inline Sources/InfluxDBSwift/FluxCSVParser.swift:144]: Following code is little bit complicated... What do you think about something like?

```swift
var recordRow: [Any] = Array()
let values: [String: Decodable] = table.columns.reduce(into: [String: Decodable]()) { result, column in
    var value: String = row[column.index + 1]
    if value.isEmpty {
        value = column.defaultValue
    }
    switch column.dataType {
    case "boolean":
        result[column.name] = (value as NSString).boolValue
    case "unsignedLong":
        result[column.name] = UInt64(value)
    case "long":
        result[column.name] = Int64(value)
    case "double":
        result[column.name] = Double(value)
    case "string":
        result[column.name] = value
    case "dateTime:RFC3339", "dateTime:RFC3339Nano":
        result[column.name] = CodableHelper.dateFormatter.date(from: value)
    case "base64Binary":
        result[column.name] = Data(base64Encoded: value)
    case "duration":
        result[column.name] = Int64(value)
    default:
        result[column.name] = value
    }
    recordRow.append(result[column.name] as Any)
}
```

**Review by bednar — CHANGES_REQUESTED** [review]
Thanks for your PR 👍 
There are a few requirements that must be be satisfy before we accept the PR:

**Review by bednar — APPROVED** [review]
LGTM 🌹
